src/coreclr/hosts/corerun/corerun.cpp

```diff
@@ -470,6 +470,18 @@ bool TryRun(const int argc, const wchar_t* argv[], Logger &log, const bool verbo
         return false;
     }
 
+    StackSString tpaList;
+    if (!managedAssemblyFullName.IsEmpty())
+    {
+        // Target assembly should be added to the tpa list. Otherwise corerun.exe
+        // may find wrong assembly to execute.
+        // Details can be found at https://github.com/dotnet/coreclr/issues/5631
+        tpaList = managedAssemblyFullName;
```

Comment:
IMO the path has already been translate to the full path at by calling [WszGetFullPathName](https://github.com/mazong1123/coreclr/blob/75406dc96e96c38627e246d65712d7d3beaa2460/src/coreclr/hosts/corerun/corerun.cpp#L397) previously. 

For Unix version, we have the similar logic at [coreconsole.cpp](https://github.com/dotnet/coreclr/blob/551a7c5563105f7ec9a9d9273204d0a07a02c4fa/src/coreclr/hosts/unixcoreconsole/coreconsole.cpp#L98) and [osxbundlerun.cpp](https://github.com/dotnet/coreclr/blob/32f0f9721afb584b4a14d69135bea7ddc129f755/src/coreclr/hosts/osxbundlerun/osxbundlerun.cpp#L33)